src/test/lib/optimizer/strategy/data_induced_predicate_rule_test.cpp

```diff
@@ -0,0 +1,128 @@
+#include "lib/optimizer/strategy/strategy_base_test.hpp"
+
+#include "logical_query_plan/aggregate_node.hpp"
+#include "logical_query_plan/join_node.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/projection_node.hpp"
+#include "optimizer/strategy/data_induced_predicate_rule.hpp"
+
+namespace hyrise {
+
+using namespace expression_functional;  // NOLINT(build/namespaces)
+
+class DataInducedPredicateRuleTest : public StrategyBaseTest {
+ protected:
+  void SetUp() override {
+    {
+      const auto histogram_column_a = GenericHistogram<int32_t>::with_single_bin(1, 50, 40, 40);
+      _node_a = create_mock_node_with_statistics({{DataType::Int, "a"}}, 40, {histogram_column_a});
+      _a_a = _node_a->get_column("a");
+    }
+
+    {
+      const auto histogram_column_a = GenericHistogram<int32_t>::with_single_bin(10, 20, 10, 10);
+      _node_b = create_mock_node_with_statistics({{DataType::Int, "a"}}, 10, {histogram_column_a});
+      _b_a = _node_b->get_column("a");
+    }
+
+    {
+      const auto histogram_column_a = GenericHistogram<int32_t>::with_single_bin(1, 50, 40, 40);
+      _node_c = create_mock_node_with_statistics({{DataType::Int, "a"}}, 40, {histogram_column_a});
+      _c_a = _node_c->get_column("a");
+    }
+  }
+
+  std::shared_ptr<MockNode> _node_a, _node_b, _node_c;
+  std::shared_ptr<LQPColumnExpression> _a_a, _b_a, _c_a;
+  std::shared_ptr<DataInducedPredicateRule> _rule{std::make_shared<DataInducedPredicateRule>()};
+};
+
+class DataInducedPredicateRuleTestJoinModeTest : public DataInducedPredicateRuleTest,
+                                                 public ::testing::WithParamInterface<JoinMode> {};
+
+INSTANTIATE_TEST_SUITE_P(JoinToPredicateRewriteRuleJoinModeTestInstance, DataInducedPredicateRuleTestJoinModeTest,
+                         ::testing::ValuesIn(magic_enum::enum_values<JoinMode>()), enum_formatter<JoinMode>);
+
+TEST_P(DataInducedPredicateRuleTestJoinModeTest, PerformRewriteLeftSide) {
+  // The rule should only rewrite inner semi, left and right joins.
+
+  // The _a_a side of the join has values from 1-50, the _b_a side has values from 10-20. Based on that
+  // selectivity, a data induced predicate should be created.
+  const auto& join_type = GetParam();
+
+  if (join_type == JoinMode::Cross) {
+    EXPECT_ANY_THROW(JoinNode::make(join_type, equals_(_a_a, _b_a), _node_a, _node_b));
+    return;
+  }
```

Comment:
I'm not sure how the JoinNode's assertion that a cross join has no predicate is particularly related to your code. Did you just mean to skip this parameter using `GTEST_SKIP();`? As an alternative, you could also provide only the desired join modes in the test suite instantiation.